duoapi.go

```diff
@@ -63,8 +72,21 @@ type DuoApi struct {
 	skey       string
 	host       string
 	userAgent  string
-	apiClient  *http.Client
-	authClient *http.Client
+	apiClient  HttpClient
+	authClient HttpClient
+	sleepSvc   SleepService
+}
+
+type HttpClient interface {
```

Comment:
We don't publicly expose the struct fields which make use of these interfaces. Maybe these should be private as well.